integration_tests/CMakeLists.txt

```
@@ -546,6 +570,7 @@ RUN(NAME structs_22          LABELS cpython llvm c NOFAST)
 RUN(NAME structs_23          LABELS cpython llvm c NOFAST)
 RUN(NAME structs_24          LABELS cpython llvm c)
 RUN(NAME structs_25          LABELS cpython llvm c)
+RUN(NAME symbolics_01        LABELS cpython sympy)
```

Let's do it like this:
```suggestion
RUN(NAME symbolics_01        LABELS cpython-sym c-sym)
```
Where `cpython-sym` is a cpython test that uses sympy (so it will only run at the CI with sympy included, otherwise it won't run). And `c-sym` is a C test with `symengine` included. Later we'll add `llvm-sym`, which will run an LLVM test with sympy included.

---

The existing names use `_` (for example `wasm_x64`, `c_py`). So, maybe we can do `cpython_sym` and `c_sym`.

---

So I'll have to introduce these in the backend right ? I guess here 
```
SUPPORTED_BACKENDS = ['llvm', 'c', 'wasm', 'cpython', 'x86', 'wasm_x86', 'wasm_x64', 'c_py', 'cpython_py', 'sympy']
```

---

Yes. Also, I suggest to use the names `cpython_sym` and `c_sym`. (These names suggest which backend it uses and also which library is uses. Also `cpython_sym` is distinct from `cpython`, hence tests using `sympy` would not be run locally when someone uses `./run_tests.py -b cpython`.)

---

Yes I do understand what You're trying to convey . Though I haven't really worked with CMake files before hence not fully sure on how I could introduce the blocks for `cpython_sym ` and `c_sym` as follows
```
elseif(KIND STREQUAL "cpython_sym ")
elseif(KIND STREQUAL "c_sym")
```
But yeah I realize that they would be closely related to how the cpython and c backends have been handled . Will try to figure it out . Any suggestions are welcome !

---

The block for `c_sym` which involves a C test including symengine , could look like the following
```
        elseif(KIND STREQUAL "c_sym")
            add_custom_command(
                OUTPUT ${name}.c
                COMMAND ${LPYTHON} ${extra_args} --show-c ${CMAKE_CURRENT_SOURCE_DIR}/${file_name}.py > ${name}.c
                DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${file_name}.py
                VERBATIM)
            add_executable(${name} ${name}.c ${extra_files})
            target_include_directories(${name} PRIVATE ${CMAKE_SOURCE_DIR}
                "${Python_INCLUDE_DIRS}/..")
            set_target_properties(${name} PROPERTIES LINKER_LANGUAGE C)
            if (APPLE)
                set(SYMPY_LIB "${Python_LIBRARY_DIRS}/libsymengine.dylib")
            else()
                set(SYMPY_LIB "${Python_LIBRARY_DIRS}/libsymengine.so")
            endif()
            target_link_libraries(${name} lpython_rtlib ${SYMPY_LIB})
            add_test(${name} ${CMAKE_CURRENT_BINARY_DIR}/${name})
            if (labels)
                set_tests_properties(${name} PROPERTIES LABELS "${labels}")
            endif()
            if (${fail})
                set_tests_properties(${name} PROPERTIES WILL_FAIL TRUE)
            endif()
 ```

---

Yes, it looks good.

For `cpython_sym`, I think for now you can just do `elseif((KIND STREQUAL "cpython") OR (KIND STREQUAL "cpython_py") OR (KIND STREQUAL "cpython_sym"))`.